kubeflow/trainer/rhai/traininghub_instrumentation.py

```
@@ -0,0 +1,961 @@
+        metrics_file_rank0=metrics_file_rank0,
+    )
+
+    # Wrap the call in if __name__ == '__main__' to prevent re-execution
+    # when algorithms use multiprocessing.spawn (e.g. ART's lora_grpo).
+    # Harmless for algorithms that don't use spawn (SFT, OSFT, LoRA SFT).
+    if func_args is None:
+        call_line = "if __name__ == '__main__':\n    training_func({})\n"
+    elif isinstance(func_args, dict):
+        params_lines: list[str] = ["if __name__ == '__main__':\n    training_func({\n"]
+        for key, value in func_args.items():
+            params_lines.append(f"        {repr(key)}: {repr(value)},\n")
```

Ruff (0.16.0)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)